ZFSin/zfs/module/zfs/zfs_windows_zvol_scsi.c

```diff
@@ -792,28 +808,42 @@ wzvol_WkRtn(__in PVOID pWkParms)                          // Parm list pointer.
 		goto Done;
 	}
 
-
 	if (sectorOffset >= zv->zv_volsize) {      // Starting sector beyond the bounds?
 		dprintf("%s: invalid starting sector: %d\n", __func__, startingSector);
 		status = SRB_STATUS_INVALID_REQUEST;
 		goto Done;
 	}
 
+	InterlockedIncrement((LONG volatile *)&zv->zv_requests_processing_count);
 	/* Call ZFS to read/write data */
-	if (ActionRead == pWkRtnParms->Action) {           
+	if (ActionRead == pWkRtnParms->Action) {
 		status = zvol_read_win(zv, sectorOffset, pSrb->DataTransferLength, pX);
-	} else {                                           
+	} else {
 		status = zvol_write_win(zv, sectorOffset, pSrb->DataTransferLength, pX);
 	}
 
-	if (status == 0)
-		status = SRB_STATUS_SUCCESS;
+	if (status == 0) {
+		/* Both function return status = 0, when we have request cancellation */
+		request_cancellation = InterlockedCompareExchange((LONG volatile *)&zv->zv_request_cancellation, 0, 0);
```

Comment:
Using the Windows `InterlockedCompareExchange()` is OK here, since we are in the Windows only source files...